hack/tools/logcheck/main.go

```diff
@@ -27,14 +28,26 @@ import (
 	"golang.org/x/tools/go/analysis/singlechecker"
 )
 
+var logcheckFlags = flag.NewFlagSet("", flag.ExitOnError)
```

Comment:
Let's avoid adding more globals, could you move Analyser and flag registration to functions?
I know globals are simpler, but they create a bad pattern in code. Let me know if you need help with this.

Reply:
Discussion here: https://github.com/kubernetes/klog/pull/247#discussion_r659618533